### DISCUSSION TITLE
Add Update, Delete in SQL section from tutorial

### DISCUSSION THREAD

**Discussion by akihiro-inui** [fastapi/fastapi#8996] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * Or, I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### New Feature
I have added the following examples to the documentation under SQL section.
- Update operation with database
- Delete operation with database
- UPDATE endpoint
- DELETE endpoint

### Pull Request
Please see this pull request to review the exact changes.
https://github.com/tiangolo/fastapi/pull/2665

**Comment by mikey-no** [comment]
Like the enhancement. I have just spotted a small mistake.. [CRUD comes from: Create, Read, Update, and Delete.](https://6008a3f7e319bf0ea83a17ce--fastapi.netlify.app/tutorial/sql-databases/#crud-utils)
Next it says: "...although in this example we are only creating and reading."
This line should be deleted.

**Comment by jsun969** [comment]
Your document help me a lot but there may be a problem in **delete**
```diff python
 def delete_user(db: Session, user_id: int):
     db_user_to_delete = db.query(models.User).filter(models.User.id == user_id).first()
     db.delete(db_user_to_delete)
     db.commit()
-    db.refresh(db_user_to_delete)
     return db_user_to_delete
```
the delete instance can not be refresh as it is not persistent

**Comment by mikey-no** [comment]
jsun969, 

I see the problem. When an item has been deleted you cannot return it. 

I would suggest returning a HTTP response code of 204... "a 204 (No Content) status code if the action has been enacted and no further information is to be supplied" [RFC 7321 Section 4.3.5.  DELETE](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.5).

I have not found best way to do this (yet). I found this: [[QUESTION] How to send 204 response?#717](https://github.com/tiangolo/fastapi/issues/717)

From the above question I would take this as the answer:
`return Response(status_code=HTTP_204_NO_CONTENT)`

This page may also help [about-http-status-codes](https://fastapi.tiangolo.com/tutorial/response-status-code/#about-http-status-codes).

Also see this [Empty Response Body .... ](https://swagger.io/docs/specification/describing-responses/)

Still no complete answer :|

Thanks

mikey-no

**Comment by YuriiMotov** [comment]
Done: https://fastapi.tiangolo.com/tutorial/sql-databases/
